pkg/drivers/vmware/driver.go

```diff
@@ -401,6 +406,18 @@ func (d *Driver) getMacAddressFromVmx() (string, error) {
 	return macaddr, nil
 }
 
+func (d *Driver) getIPfromVmrun() (string, error) {
+	vmx := d.vmxPath()
+
+	ip := regexp.MustCompile(`\d+\.\d+\.\d+\.\d+`)
+	stdout, _, _ := vmrun("getGuestIPAddress", vmx)
```

Comment:
@middagj this doesn't seem to work for me, due to https://communities.vmware.com/t5/VMware-Fusion-Discussions/Using-vmrun-to-obtain-a-guests-IP-address-when-running-the-VM-in/td-p/2236978
It appears to be an issue with `getGuestIPAddress` not working correctly in VMWare headless mode.

Specifically, it fails with the very odd error `Error: The VMware Tools are not running in the virtual machine: <path_to_vmx>` eventually exhausting all retries, unless you open the VMs UI.

The fix (which is also bizarre) appears to be to add in a `-wait` as described in that issue. I've tested that with your changes and it then immediately gets the IP after the VM is started. I'm not sure how appropriate that is since there are other fallback methods to get the IP; perhaps it needs to be paired with a timeout of some sort.

```suggestion
	stdout, _, _ := vmrun("getGuestIPAddress", vmx, "-wait")
```

Reply:
I think I sometimes experienced this when I tried to get the docker-env variables in a shell. I added the suggestion along with a maximum wait. I put a default for 30s, should be enough I guess, and made it configurable. Not sure about the naming. Also not thoroughly tested. Does this solves the problem for you? Any suggestion on a proper default?

Very disappointing that the issue is open for almost 5 years now.

Reply:
Yes, that appears to work just fine for me.

Tangentially related, unfortunately even with this change I now hit https://github.com/kubernetes/minikube/issues/3036 on VMWare since the new default "Internet Sharing" network mode conflicts with a dns proxy server on the host. It's unfortunate VMWare made such a breaking change here. The good news though is that VMWare does still appear to have a bridged networking mode (that still uses `bridge100` interface like sharing also seems to). That mode also seems to work fine with `vmrun getGuestIPAddress`, but it doesn't look like `docker-machine-driver-vmware` supports selecting the network mode on creation so I might have another PR coming this way...

Reply:
I expect `-wait` to exit early, so better be on the safe side.

Isn't "Internet Sharing" always been the default with Fusion? I found it the most convenient with companies that usually have a VPN. That brings me to an additional problem I was facing after the upgrade to Big Sur. The network on the host wasn't fully accessible even with "Internet Sharing" on. I am using the solution of https://communities.vmware.com/t5/VMware-Fusion-Discussions/Share-host-VPN-with-guest/m-p/2810095/highlight/true#M170151

```zsh
#!/usr/bin/env zsh

# Source: https://communities.vmware.com/t5/VMware-Fusion-Discussions/Share-host-VPN-with-guest/m-p/2810095/highlight/true#M170151

set -e

if [[ -z $1 ]]; then
    echo >&2 "Specify interface"
    exit 1
fi
iface="$1"

if ! ifconfig | grep -qe "$iface"; then
    echo >&2 "Interface $iface does not exist"
    exit 1
fi

nat="$(sudo pfctl -a com.apple.internet-sharing/shared_v4 -s nat 2> /dev/null)"
if grep -qe "^nat on $iface inet from" <<< "$nat"; then
    echo >&2 "Interface $iface already added"
    exit
fi

src="$(sed -nE 's#^nat on en[0-9]+ inet from (.+) to any.*$#\1#p' <<< "$nat" | head -n1)"
if [[ -z $src ]]; then
    echo >&2 "Could not find source rule"
    exit 1
fi

nat="$nat"$'\n'"nat on $iface inet from $src to any -> ($iface) extfilter ei"
echo $nat
sudo pfctl -a com.apple.internet-sharing/shared_v4 -N -f - <<< "$nat" 2>/dev/null
```

It was apparently in the Release Notes, workaround is to connect on the machine which isn't really doable with minikube.